app/controllers/api/v1/attachments_controller.rb

```diff
@@ -19,8 +17,19 @@ class AttachmentsController < OpenStax::Api::V1::ApiController
       #{json_schema(Api::V1::AttachmentRepresenter, include: :readable)}
     EOS
     def create
+      exercise = Exercise
+                   .visible_for(user: current_api_user)
+                   .with_id(params[:exercise_id]).first
+      if !exercise && params[:exercise_id] =~ /(\d+)@draft/
+        published = Exercise
+                     .visible_for(user: current_api_user)
+                     .with_id($1).first!
+        exercise = published.new_version
+        exercise.save!
+      end
```

Comment:
I think it might be better to reuse the code in https://github.com/openstax/exercises/blob/08d77274be38e99ffb574b21462e9a71f984a505/app/controllers/api/v1/exercises_controller.rb#L189 since it deals with locking the exercise and permission checks (maybe move it into a module so it can be mixed in?)